Documentation/upgrade.md

```diff
@@ -4,47 +4,68 @@ weight: 60
+With this upgrade guide, there are a few notes to consider:
+* **WARNING:** Upgrading a Rook cluster is not without risk. There may be unexpected issues or
+  obstacles that damage the integrity and health of your storage cluster, including data loss. Only
+  proceed with this guide if you are comfortable with that.
+* Rook is able to handle a great deal of the upgrade on its own, but manual steps are still
+  necessary. This is in part to reduce the Kubernetes privileges required by the Rook operator.
+* This guide assumes that your Rook operator and its agents are running in the `rook-system`
+  namespace. It also assumes that your Rook cluster is in the `rook` namespace. If any of these
+  components is in a different namespace, search/replace all instances of `-n rook-system` and
+  `-n rook` in this guide with `-n <your namespace>`.
+  * New Ceph specific namespaces (`rook-ceph-system` and `rook-ceph`) are now used by default in the
+    new release, but this guide maintains the usage of `rook-system` and `rook` for backwards
+    compatibility. Note that all user guides and examples have been updated to the new namespaces,
+    so you will need to tweak them to maintain compatibility with the legacy `rook-system` and
+    `rook` namespaces.
 
 ## Prerequisites
 In order to successfully upgrade a Rook cluster, the following prerequisites must be met:
 * The cluster should be in a healthy state with full functionality.
-Review the [health verification section](#health-verification) in order to verify your cluster is in a good starting state.
-* `dataDirHostPath` must be set in your Cluster spec.
-This persists metadata on host nodes, enabling pods to be terminated during the upgrade and for new pods to be created in their place.
-More details about `dataDirHostPath` can be found in the [Cluster CRD readme](./ceph-cluster-crd.md#cluster-settings).
-* All pods consuming Rook storage should be created, running, and in a steady state.  No Rook persistent volumes should be in the act of being created or deleted.
-  * If you are using rook release `v0.8.1` or older, you might have PVCs stuck in pending state. The issue is tracked [here](https://github.com/rook/rook/issues/2072). One of the two suggestions below might be a path forward.
-    * Upgrade operator to the latest release. On restart, track the logs to verify that PVCs are transitioned out of pending to bound state.
-    * Delete the pending PVCs and re-create them after the rook cluster has been upgraded.
+  Review the [health verification section](#health-verification) in order to verify your cluster is
+  in a good starting state.
+* `dataDirHostPath` must be set in your Cluster spec. This persists metadata on host nodes, enabling
```

Comment:
as of v0.8, `dataDirHostPath` should be required already, so you can remove this statement.